sdk/core/Azure.Core.TestFramework/src/TestRecording.cs

```diff
@@ -51,7 +51,7 @@ public TestRecording(RecordedTestMode mode, string sessionFile, RecordedTestSani
                     }
                     catch (Exception ex) when (ex is FileNotFoundException || ex is DirectoryNotFoundException)
                     {
-                        throw new TestRecordingMismatchException(ex.Message, ex);
+                        _mismatchException = new TestRecordingMismatchException(ex.Message, ex);
```

Comment:
Should we call load in the Session getter instead?

Reply:
I like it.

Reply:
hmm, actually there is one scenario where this doesn't work. `HasRequests` needs to not trigger `Load` but also be accurate after the ctor runs. This change would make that tricky.

Reply:
`HasRequests` is only used during recording so it might be okayish not to trigger load in it are return false if nothing is loaded.

Reply:
Although it happens to be true today, I'd be concerned that it could cause an issue if anyone tried to rely on `HasRequests` for any other reason.

Reply:
But in the current implementation, it also returns the wrong value when there is no Session loaded. In theory, it should throw.

Reply:
Isn't `false` still correct in that case? Perhaps we should add an additional `SessionLoaded` bool so that one could get the full picture if that is what is intended? I'm not sure that having to wrap `HasRequests` in a try/catch is better.

Reply:
Yeah, maybe. I just don't like relying on catching and suppressing exceptions.